stellar_rust_sdk/src/trade_aggregations/trade_aggregations_request.rs

```diff
@@ -0,0 +1,374 @@
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing either the updated `TradeAggregationsRequest` or an error.
+    ///
+    pub fn set_offset(self, offset: u64) -> Result<Self, String> {
+        const ONE_HOUR: &u64 = &360000;
+        const ONE_DAY: &u64 = &86400000;
+        let resolution = format!("{}", &self.resolution.0)
+            .parse::<u64>()
+            .unwrap();
+
+        let conditions = [
+            (&resolution < ONE_HOUR, "Resolution must be greater than 1 hour when setting offset."),
+            (&offset % ONE_HOUR != 0, "Offset must be in whole hours."),
+            (&offset > &resolution, "Offset must be smaller than the resolution."),
+            (&offset > ONE_DAY, "Offset must be smaller than 24 hours."),
+        ];
+
+        for (condition, message) in conditions {
+            if condition {
+                return Err(message.to_string())
+            }
+        }
+
+        Ok(Self {
+            offset: Some(offset.to_string()),
+            ..self
+        })
+    }
+}
+
+
+impl Request for TradeAggregationsRequest<BaseAsset, CounterAsset, Resolution> {
+    fn get_query_parameters(&self) -> String {
+        let mut asset_parameters: Vec<String> = Vec::new();
+        match &self.base_asset.0 {
+            AssetType::Native => {
+                asset_parameters.push(format!("base_asset_type=native"));
+            }
+            AssetType::Alphanumeric4(asset) => {
+                asset_parameters.push(format!("base_asset_type=credit_alphanum4"));
+                asset_parameters.push(format!("&base_asset_code={}", asset.asset_code));
+                asset_parameters.push(format!("&base_asset_issuer={}", asset.asset_issuer));
+            }
+            AssetType::Alphanumeric12(asset) => {
+                asset_parameters.push(format!("base_asset_type=credit_alphanum12"));
+                asset_parameters.push(format!("&base_asset_code={}", asset.asset_code));
+                asset_parameters.push(format!("&base_asset_issuer={}", asset.asset_issuer));
+            }
+        }
+
+        match &self.counter_asset.0 {
+            AssetType::Native => {
+                asset_parameters.push(format!("&counter_asset_type=native"));
+            }
+            AssetType::Alphanumeric4(asset) => {
+                asset_parameters.push(format!("&counter_asset_type=credit_alphanum4"));
+                asset_parameters.push(format!("&counter_asset_code={}", asset.asset_code));
+                asset_parameters.push(format!("&counter_asset_issuer={}", asset.asset_issuer));
+            }
+            AssetType::Alphanumeric12(asset) => {
+                asset_parameters.push(format!("&counter_asset_type=credit_alphanum12"));
+                asset_parameters.push(format!("&counter_asset_code={}", asset.asset_code));
+                asset_parameters.push(format!("&counter_asset_issuer={}", asset.asset_issuer));
+            }
+        }
```

Comment:
Deze constructie heb ik eerder gezien. Er zit veel duplicate code bij.
Kun je m in lijn met deze implementatie brengen?

https://github.com/Baseflow/stellar-rust-sdk/blob/898b80bdc83f13c221e910d8cfc057c13cb8e718/stellar_rust_sdk/src/liquidity_pools/all_liquidity_pools_request.rs#L138

Reply:
Klopt, deze komt elders ook voor. Ik dacht dat we hier een aparte issue voor hadden aangemaakt (zodat ik dit in de maintenance-fase kan oppakken) maar deze staat er nog niet bij. Zal ik 'm nu meenemen, of in een nieuwe feature branch? Dan zou ik gelijk die andere implementatie kunnen refactoren.